apps/bestofjs-nextjs/src/server/api-projects.tsx

```diff
@@ -0,0 +1,148 @@
+import * as mingo from "mingo";
+import { RawObject } from "mingo/types";
+
+import { filterProjectsByQuery } from "@/components/search-palette/find-projects";
```

Comment:
I'm looking at `find-projects.ts`, I'm a little surprised this comes from `@/components`

It seems like a util function mostly used by `search-palette`

Maybe it would be more clear to me if the file structure was changed.

`components/search-palette/utils/find-projects`

Reply:
Good catch! We need a kind of `shared` folder for functions used by both components and the API layer, as the server should not import functions coming from components.

Reply:
Done here: https://github.com/bestofjs/bestofjs/pull/200/commits/4e3d1ba28ef9299202186586631729e6c96f376a